src/taipy/core/data/data_node.py

```diff
@@ -181,10 +181,13 @@ def validity_period(self, val):
     @property  # type: ignore
     @_self_reload(_MANAGER_NAME)
     def expiration_date(self) -> datetime:
-        if not self._last_edit_date:
+        last_edit_date = self.last_edit_date
```

Comment:
I want to improve this part since we're reloading the dn twice here, any ideas would help :D

Reply:
I don't have any idea at this point.

Reply:
Can we remove the `@_self_reload(_MANAGER_NAME)` wrapper here? Since `last_edit_date = self.last_edit_date` will reload the data node.

Reply:
we certainly can, but it doesn't help improve the readability of the code. By that I mean we are calling self.last_edit_date while also calling self._validity_period. Can be confusing for others to understand